index.js

```diff
@@ -3,60 +3,72 @@ const fs = require('fs-extra');
 const path = require('path');
 
 const run = (config, {logger}) => {
+    if (!Array.isArray(config.files)) {
+        return Promise.resolve({
+            status: 'error',
+            message: 'Configuration does not contain a valid "files" property.'
+        });
+    }
 
-	if(!Array.isArray(config.files)) {
-		return Promise.resolve({
-			status: 'error',
-			message: 'Configuration does not contain a valid "files" property.'
-		});
-	}
-
-	const promises = config.files.map(fileConfig => processFile(fileConfig, config.plugins, logger));
-
-	return Promise.all(promises)
-		.then(() => {
-			const message = `${config.files.length} stylesheet(s) processed.`;
-			logger.info(message);
-			return {
-				status: 'complete',
-				message: message
-			};
-		})
-		.catch(e => {
-			logger.error(e);
-			return {
-				status: 'error',
-				message: e.message
-			};
-		});
+    const promises = config.files.map(fileConfig => processFile(fileConfig, config.plugins, logger));
 
+    return Promise.all(promises)
+        .then(() => {
+            const message = `${config.files.length} stylesheet(s) processed.`;
+            logger.info(message);
+            return {
+                status: 'complete',
+                message: message
+            };
+        })
+        .catch(e => {
+            logger.error(e);
+            return {
+                status: 'error',
+                message: e.message
+            };
+        });
 };
 
 const processFile = (fileConfig, plugins, logger) => {
-	const srcFilepath = path.resolve(process.cwd(), fileConfig.src);
-	const destFilepath = path.resolve(process.cwd(), fileConfig.dest);
-
-	return fs.readFile(srcFilepath)
-		.then(contents =>
-			postcss(plugins).process(contents, {
-				from: srcFilepath,
-				to: destFilepath
-			})
-		)
-		.then(result => {
-			displayWarnings(result, logger);
-			return fs.ensureDir(path.dirname(destFilepath))
-				.then(() => fs.writeFile(destFilepath, result.css));
-		});
+    const srcFilepath = path.resolve(process.cwd(), fileConfig.src);
+    const destinationArray = getDestinations(fileConfig);
+    const params = {plugins, logger, srcFilepath, destinationArray};
+    const promises = initOutput(params);
+    return Promise.all(promises);
+};
+
+const getDestinations = fileConfig => {
+    if (Array.isArray(fileConfig.dest)) {
```

Comment:
Isn't this conditional logic backward?

Reply:
Now that I think about..yes. But are there any caveats to this? The results are the same.

Reply:
I wonder if maybe your test just isn't catching the possible issue. If you flip the logic, it should be ok -- you're basically just ensuring that you always have an array...I think.

Reply:
If I flip the logic, I am then using the spread operator on a string,  turning it into an array of characters. That's why it does not seem to work and the tests now fail. Am I just missing something here?

Reply:
Ohhh, sorry, yeah, it was fine all along, ignore me :)

Reply:
Haha you took me for a ride there.